sns/utils.bal

```diff
@@ -0,0 +1,284 @@
+isolated function setSubscriptionAttributes(map<string> parameters, SubscriptionAttribute attributes) returns map<string> {
+    int attributeNumber = 1;
+    map<anydata> attributeMap = <map<anydata>>attributes;
+    foreach var [key, value] in attributeMap.entries() {
+        string attributeName = getAttributeName(key);
+        parameters["Attributes.entry." + attributeNumber.toString() + ".Name"] = attributeName.toString();
+        parameters["Attributes.entry." + attributeNumber.toString() + ".Value"] = value.toString();
+        attributeNumber = attributeNumber + 1;
+    }
+    return parameters;
+}
+
+# Set message attributes to a map of string to add as query parameters.
+#
+# + parameters - Parameter map
+# + attributes - MessageAttribute to convert to a map of string
+# + return - If successful returns `map<string>` response. Else returns error
+isolated function setMessageAttributes(map<string> parameters, MessageAttribute attributes) returns map<string> {
+    int attributeNumber = 1;
+    map<anydata> attributeMap = <map<anydata>>attributes;
+    foreach var [key, value] in attributeMap.entries() {
+        string attributeName = getAttributeName(key);
+        parameters["Attributes.entry." + attributeNumber.toString() + ".Name"] = attributeName.toString();
+        parameters["Attributes.entry." + attributeNumber.toString() + ".Value"] = value.toString();
+        attributeNumber = attributeNumber + 1;
+    }
+    return parameters;
+}
+
+# Set SMS attributes to a map of string to add as query parameters.
+#
+# + parameters - Parameter map
+# + attributes - SmsAttribute to convert to a map of string
+# + return - If successful returns `map<string>` response. Else returns error
+isolated function setSmsAttributes(map<string> parameters, SmsAttribute attributes) returns map<string> {
+    int attributeNumber = 1;
+    map<anydata> attributeMap = <map<anydata>>attributes;
+    foreach var [key, value] in attributeMap.entries() {
+        string attributeName = getAttributeName(key);
+        parameters["Attributes.entry." + attributeNumber.toString() + ".Name"] = attributeName.toString();
+        parameters["Attributes.entry." + attributeNumber.toString() + ".Value"] = value.toString();
+        attributeNumber = attributeNumber + 1;
+    }
+    return parameters;
+}
+
+# Add SMS attributes.
+#
+# + parameters - Parameter map
+# + attributes - Array of attributes to convert to a map of string
+# + return - If successful returns `map<string>` response. Else returns error.
+isolated function addSmsAttributes(map<string> parameters, string[] attributes) returns map<string> {
+    int attributeNumber = 1;
+    foreach var attribute in attributes {
+        parameters["attributes.member." + attributeNumber.toString()] = attribute.toString();
+        attributeNumber = attributeNumber + 1;
+    }
+    return parameters;
+}
+
+# Get attribute name from field of record.
```

Comment:
Returns attribute name from field of record. This capitalizes the first letter of the attribute.

Reply:
Ack. Will change